src/python_testing/TC_OCC_2_2.py

```diff
@@ -0,0 +1,129 @@
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OCC_2_2(MatterBaseTest):
+    async def read_occ_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.OccupancySensing
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_OCC_2_2(self) -> str:
+        return "[TC-OCC-2.2] OccupancySensorTypeBitmap and OccupancySensorType interdependency with server as DUT"
+
+    def steps_TC_OCC_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read OccupancySensorType attribute with repect to featureMap."),
+            TestStep(3, "Read OccupancySensorTypeBitmap attribute with repect to featureMap.")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_2(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        feature_map_PIR = Clusters.OccupancySensing.Bitmaps.Feature.kPassiveInfrared
+        feature_map_US = Clusters.OccupancySensing.Bitmaps.Feature.kUltrasonic
+        feature_map_PHY = Clusters.OccupancySensing.Bitmaps.Feature.kPhysicalContact
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+            feature_map = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.attributes.FeatureMap)
+
+            if (feature_map != feature_map_PIR) & (feature_map != feature_map_US) & (feature_map != feature_map_PHY):
+                asserts.assert_equal(occupancy_sensor_type_dut,
+                                     Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir, "OccupancySensorType is not PIR")
+            elif (feature_map == feature_map_PIR) & (feature_map != feature_map_US) & (feature_map != feature_map_PHY):

```

Comment:
Please condense these somehow. I am looking at this and above for example then have an identical assert for the same conditions except a != and == difference for feature_map_pir ... so that feature map value does not matter if it results in the same assert being run and the conditions beyond the feature_map_pir are the same.

Consider maybe figuring out what the equality should be and have a single assert. Also the logic has to be explained by comments since I really cannot follow it ... why are we comparing equality on a feature map which is a bitmap?

Reply:
The whole logic overall is bad for bitmaps: if feature_map == feature_map_pir, then obviously it is not equal to the others. What you need is individual bit checks.

Reply:
Change the logic statement as requested and added comments.  Done.